source/ProcessManager.Orchestrations/InternalProcesses/MigrateCalculationsFromWholesale/V1/Activities/MigrateCalculationActivity_MigrateCalculationsFromWholesale_V1.cs

```diff
@@ -38,10 +53,129 @@ public static Guid GetMigratedWholesaleCalculationIdCustomStateGuid(Orchestratio
+            Wholesale.Model.CalculationType.ThirdCorrectionSettlement => CalculationType.ThirdCorrectionSettlement,
+            _ => throw new ArgumentOutOfRangeException(nameof(calculationType), calculationType, "Calculation type is invalid."),
+        };
+    }
+
+    private static DateTimeOffset ToDateTimeOffset(Instant instant)
+    {
+        return instant.ToDateTimeOffset();
+    }
+
+    private Task<bool> WasCalculationIdMigratedAsync(Guid wholesaleCalculationId)
+    {
+        return _processManagerContext.OrchestrationInstances
+            .AsNoTracking()
+            .Where(x => x.CustomState == GetMigratedWholesaleCalculationIdCustomState(wholesaleCalculationId))
+            .AnyAsync();
+    }
+
+    private OrchestrationInstance MigrateCalculationToOrchestrationInstance(
+        Wholesale.Model.Calculation wholesaleCalculation,
+        OrchestrationDescription brs_023_027_V1_description)
+    {
+        // Orchestration Instance => Created + Queued
+        IReadOnlyCollection<int> skipStepsBySequence = wholesaleCalculation.IsInternalCalculation
+            ? [EnqueueMessagesStep.EnqueueActorMessagesStepSequence]
+            : [];
+
+        var orchestrationInstance = _orchestrationInstanceFactory.CreateQueuedOrchestrationInstance(
+            brs_023_027_V1_description,
+            wholesaleCalculation.CreatedByUserId,
+            wholesaleCalculation.CreatedTime,
+            wholesaleCalculation.ScheduledAt,
+            skipStepsBySequence);
+
+        orchestrationInstance.CustomState.Value = $"{MigratedWholesaleCalculationIdCustomStatePrefix}{wholesaleCalculation.Id}";
+
+        var calculationInput = new CalculationInputV1(
+            CalculationType: ToCalculationType(wholesaleCalculation.CalculationType),
+            GridAreaCodes: wholesaleCalculation.GridAreaCodes.Select(x => x.Code).ToList(),
+            PeriodStartDate: ToDateTimeOffset(wholesaleCalculation.PeriodStart),
+            PeriodEndDate: ToDateTimeOffset(wholesaleCalculation.PeriodEnd),
+            IsInternalCalculation: wholesaleCalculation.IsInternalCalculation);
+        orchestrationInstance.ParameterValue.SetFromInstance(calculationInput);
+
+        // Orchestration Instance => Running
+        orchestrationInstance.Lifecycle.TransitionToRunning(new MagicClock(wholesaleCalculation.ExecutionTimeStart));
+
+        // Step: Calculation
+        orchestrationInstance.TransitionStepToRunning(1, new MagicClock(wholesaleCalculation.ExecutionTimeStart));
+        orchestrationInstance.TransitionStepToTerminated(1, OrchestrationStepTerminationState.Succeeded, new MagicClock(wholesaleCalculation.ExecutionTimeEnd));
+
+        // Step: Enqueue messages
+        var stepsSkippedBySequence = orchestrationInstance.Steps
+            .Where(step => step.IsSkipped())
+            .Select(step => step.Sequence)
+            .ToList();
+        if (!stepsSkippedBySequence.Contains(2))
```

Comment:
```suggestion
        if (!stepsSkippedBySequence.Contains(EnqueueMessagesStep.EnqueueActorMessagesStepSequence))
```
Same for the "2"'s below.

Reply:
Won't fix. The plan is to delete the code within 14 days